### PR TITLE
Support annotation.textcoords == "offset points"

### PR DESCRIPTION
**Pull request by eric-wieser**
This modifies `_annotation` to return the location at which the text should be rendered, and then adds support for `offset points` and `offset pixels`.

This should make it easier to add support for the other coordinate type too, but I'll leave that for someone who needs them.

### CODE REVIEW

File: tikzplotlib/_text.py [hunk 1]
```diff
@@ -9,18 +9,32 @@ def draw_text(data, obj):
...
-        _annotation(obj, data, content)
+        pos = _annotation(obj, data, content)
+    else:
+        pos = obj.get_position()
 
-    # 1: coordinates
-    # 2: properties (shapes, rotation, etc)
-    # 3: text style
-    # 4: the text
-    #                   -------1--------2---3--4--
```
Comment (eric-wieser) [inline tikzplotlib/_text.py]: No idea what was intended by this comment, but it seems to have lost its usefulness

File: tikzplotlib/_text.py [hunk 2]
```diff
@@ -140,58 +139,91 @@ def _transform_positioning(ha, va):
...
+                data, col, _ = _color.mpl_color2xcolor(
+                    data, obj.arrow_patch.get_ec()
+                )
+                arrow_style.append(col)
+
+    if arrow_style:
+        the_arrow = (
+            "\\draw[{}] {} -- {};\n"
+        ).format(",".join(arrow_style), text_pos, xy_pos)
         content.append(the_arrow)
```
Comment (eric-wieser) [inline tikzplotlib/_text.py]: Now only draws arrows if one was requested - annotations by default do not have any line at all.

**Comment by nschloe** [comment]
You need to `make format`. Also, some of the test outputs are changing with this. Use `test/refresh_reference_files.py` to update.

**Comment by eric-wieser** [comment]
Will do that if / when I get around to actually making a git checkout, and not just editing these files in site-packages and pasting into github...

**Comment by eric-wieser** [comment]
Turns out I needed to switch to a different python installation, at which point editing `site-package` wasn't an option. Updated.

File: tikzplotlib/_text.py [hunk 3]
```diff
@@ -140,58 +138,89 @@ def _transform_positioning(ha, va):
...
+        raise NotImplementedError
+    elif coords == "axes fraction":
+        raise NotImplementedError
+    elif coords == "data":
+        raise NotImplementedError
+    elif coords == "polar":
+        raise NotImplementedError
+    else:
+        # unknown
+        raise NotImplementedError
```
Comment (eric-wieser) [inline tikzplotlib/_text.py]: Coverage failure is from these lines

Comment (nschloe) [reply tikzplotlib/_text.py]: Coverage failures aren't so bad.

**Comment by nschloe** [comment]
LGTM, thanks!

**Pull request merged**